dags/roger/core.py

```diff
@@ -782,216 +807,58 @@ def write_schema (self, schema, schema_type: SchemaType):
-                                      current_metric=current_metric,
-                                      data_set_name=file)
-
-        log.info(f"total time for dumping to redis : {time.time() - start}")
-
-        # now we have all nodes and edges merged in redis we scan the whole redis back to disk
+        jsonl_format_files = Util.kgx_objects("jsonl")
+
+        # Create lists of the nodes and edges files in both json and jsonl formats
+        json_node_files = {file for file in json_format_files if "node" in file}
+        jsonl_node_files = {file for file in jsonl_format_files if "node" in file}
+        json_edge_files = {file for file in json_format_files if "edge" in file}
+        jsonl_edge_files = {file for file in jsonl_format_files if "edge" in file}
+
+        # Create all the needed iterators and sets thereof
+        json_node_iterators = [jsonl_iter(file_name, 'nodes') for file_name in json_node_files] 
+        jsonl_node_iterators = [jsonl_iter(file_name) for file_name in jsonl_node_files] 
+        json_edge_iterators = [jsonl_iter(file_name, 'edges') for file_name in json_edge_files] 
+        jsonl_edge_iterators = [jsonl_iter(file_name) for file_name in jsonl_edge_files] 
+        all_node_iterators = json_node_iterators + jsonl_node_iterators
+        all_edge_iterators = json_edge_iterators + jsonl_edge_iterators
+
+        # chain the iterators together
+        node_iterators = chain(*all_node_iterators)
+        edge_iterators = chain(*all_edge_iterators)
+
+        # now do the merge
+        self.merger.merge_nodes(node_iterators)
+        merged_nodes = self.merger.get_merged_nodes_jsonl()
+
+        self.merger.merge_edges(edge_iterators)
+        merged_edges = self.merger.get_merged_edges_jsonl()
+
         write_merge_metric = {}
         t = time.time()
-        log.info("getting all nodes")
         start_nodes_jsonl = time.time()
         nodes_file_path = Util.merge_path("nodes.jsonl")
-        self.write_redis_back_to_jsonl(nodes_file_path, "node-*")
-        log.info(f"writing nodes to took : {time.time() - start_nodes_jsonl}")
+
+        # stream out nodes to nodes.jsonl file
+        with open(nodes_file_path, 'w', buffering=(1024*512)) as stream:
+            for nodes in merged_nodes:
+                stream.write(json.dumps(nodes).decode('utf-8') + '\n')
+
+        log.info(f"writing nodes took : {time.time() - start_nodes_jsonl}")
         write_merge_metric['nodes_writing_time'] = time.time() - start_nodes_jsonl
         start_edge_jsonl = time.time()
-        log.info("getting all edges")
-        edge_output_file_path = Util.merge_path("edges.jsonl")
-        self.write_redis_back_to_jsonl(edge_output_file_path, "edge-*")
+
+        edges_file_path = Util.merge_path("edges.jsonl")
+        with open(edges_file_path, 'w', buffering=(1024*512)) as stream:
+            for edges in merged_edges:
+                stream.write(json.dumps(edges).decode('utf-8') + '\n')
+
```

Comment:
by the end of this do we need to clean up the temp dir used for merging ?

Reply:
I just checked this.  If it runs to completion, it cleans up the files.  If I Ctr-C it, it does not. I'm unsure how this happens.